src/pages/studyView/charts/barChart/BarChart.tsx

```diff
@@ -0,0 +1,236 @@
+import * as React from "react";
+import { observer } from "mobx-react";
+import { VictoryChart, VictoryBar, VictoryAxis, VictorySelectionContainer } from 'victory';
+import { computed } from "mobx";
+import _ from "lodash";
+import CBIOPORTAL_VICTORY_THEME from "shared/theme/cBioPoralTheme";
+import {ClinicalDataIntervalFilterValue, DataBin} from "shared/api/generated/CBioPortalAPIInternal";
+import { AbstractChart } from "pages/studyView/charts/ChartContainer";
+import { bind } from "bind-decorator";
+import BarChartAxisLabel from "./BarChartAxisLabel";
+import {
+    calcIntervalBinValues, filterCategoryBins, filterIntervalBins, filterNumericalBins, generateCategoricalData,
+    generateNumericalData, isLogScaleByValues
+} from "../../StudyViewUtils";
+
+export interface IBarChartProps {
+    data: DataBin[];
+    filters: ClinicalDataIntervalFilterValue[];
+    onUserSelection: (dataBins: DataBin[]) => void;
+}
+
+export type BarDatum = {
+    x: number,
+    y: number,
+    dataBin: DataBin
+};
+
+function generateTheme() {
+    const theme = _.cloneDeep(CBIOPORTAL_VICTORY_THEME);
+    theme.axis.style.tickLabels.fontSize *= 0.85;
+
+    return theme;
+}
+
+const VICTORY_THEME = generateTheme();
+
+const OPERATOR_MAP: {[op:string]: string} = {
+    "<=": "≤",
+    "<" : "<",
+    ">=": "≥",
+    ">": ">"
+};
+
+@observer
+export default class BarChart extends React.Component<IBarChartProps, {}> implements AbstractChart {
+
+    private svgContainer: any;
+
+    constructor(props: IBarChartProps) {
+        super(props);
+    }
+
+    @bind
+    private onSelection(bars: {data: BarDatum[]}[], bounds: {x: number, y: number}[], props: any) {
+        const dataBins = _.flatten(bars.map(bar => bar.data.map(barDatum => barDatum.dataBin)));
+        this.props.onUserSelection(dataBins);
+    }
+
+    private isDataBinSelected(dataBin: DataBin, filters: ClinicalDataIntervalFilterValue[]) {
+        return filters.find(filter =>
+            (filter.start === dataBin.start && filter.end === dataBin.end) ||
+            (filter.value !== undefined && filter.value === dataBin.specialValue)
+        ) !== undefined;
+    }
+
+    public downloadData() {
+        // TODO we don't have the full data here
+        return this.props.data.map(obj => obj.specialValue + '\t' + obj.count).join('\n');
+    }
+
+    public toSVGDOMNode(): Element {
+        return this.svgContainer.firstChild;
+    }
+
+    @computed get numericalBins() {
+        return filterNumericalBins(this.props.data);
+    }
+
+    @computed get categoryBins() {
+        return filterCategoryBins(this.props.data);
+    }
+
+    @computed get numericalData(): BarDatum[] {
+        return generateNumericalData(this.numericalBins);
+    }
+
+    @computed get categoricalData(): BarDatum[] {
+        return generateCategoricalData(this.categoryBins, this.numericalTickFormat.length);
+    }
+
+    @computed get numericalTickFormat() {
+        if (this.numericalBins.length === 0) {
```

Comment:
seems prime for a test.

Reply:
We should probably first externalize this into `StudyViewUtils` and then add some tests.